File: 1528-kids-with-the-greatest-number-of-candies/solution.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max = -1;
        for (int i = 0; i < candies.length; i++) {

            if (candies[i] > max) {
                max = candies[i];
            }
            candies[i] += extraCandies;
        }
        List<Boolean> result = new ArrayList<>();
        for (int i : candies) {
            if (i >= max) {
                result.add(true);
            } else {
                result.add(false);
            }
        }
        return result;
    }
}

File: 1528-kids-with-the-greatest-number-of-candies/solution.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
    maxCandiesKidHas := 0
    for _, c := range candies {
        if c > maxCandiesKidHas {
            maxCandiesKidHas = c
        }
    }

    res := make([]bool, 0)
    for _, c := range candies {
        if c + extraCandies >= maxCandiesKidHas {
            res = append(res, true)
        } else {
            res = append(res, false)
        }
    }

    return res
}
